wbooks-api/spec/controllers/api/v1/books_controller_spec.rb

```diff
@@ -1,4 +1,53 @@
 require 'rails_helper'
+require 'controllers/api/v1/authenticated_user'
 
-# RSpec.describe Api::V1::BooksController, type: :controller do
-# end
+describe Api::V1::BooksController do
+  include_context 'Authenticated User'
+
+  describe 'GET #index' do
+    context 'when fetching all the books' do
+      let!(:books) { create_list(:book_with_rents, 3) }
+
+      before do
+        get :index
+      end
+
+      it 'responses with all the books' do
+        expected = JSON.parse(ActiveModel::Serializer::CollectionSerializer.new(
+          books, serializer: BookIndexSerializer
+        ).to_json)
+
+        content = JSON.parse(response.body)['page']
```

Comment:
we always define a helper to do this. Add this:
```
module Response
  module JSONParser
    def response_body
      ActiveSupport::JSON.decode(response.body) if response.present?
    end
  end
end
```
in `spec/support/parsed_response_helper.rb` . Also add it in `rails_helper`